extensions/amp-analytics/0.1/visibility-impl.js

```diff
@@ -321,6 +318,76 @@ export class Visibility {
     }
   }
 
+  /**
+   * @param {!Object} config
+   * @param {function(!Object)} callback
+   * @param {boolean} shouldBeVisible True if the element should be visible
+   *   when callback is called. False otherwise.
+   * @param {!Element} analyticsElement The amp-analytics element that the
+   *   config is associated with.
+   */
+  listenOnceV2(config, callback, shouldBeVisible, analyticsElement) {
+    const selector = config['selector'];
+    const element = user().assertElement(
+        getElement(this.ampdoc, selector,
+            dev().assertElement(analyticsElement),
+            config['selectionMethod']),
+        'Element not found for visibilitySpec: ' + selector);
+
+    const resource =
```

Comment:
it's needed both in V1 and V2. we will wait for resource load before adding the listener.